webapp-client/src/component/Navbar.js

```diff
@@ -0,0 +1,87 @@
+import React, { Fragment } from 'react';
+import '../stylesheets/Navbar.css';
+import Category from './Category';
+
+export default class Navbar extends React.Component {
+  constructor(props) {
+    super(props);
+    this.default = 'All';
+    this.state = {
+      categories: [this.default],
+      display: this.default,
+    };
+    this.addCategory = this.addCategory.bind(this);
+    this.setContent = this.setContent.bind(this);
+  }
+
+  addCategory(category) {
+    this.setState({
+      categories: [
+        ...this.state.categories,
+        category,
+      ]
+    });
+  }
+
+  setContent(category) {
+    this.setState({
+      display: category,
+    });
+  }
+
+  render() {
+    return (
+      <Fragment>
+        <div className="nav-bar">
+          <CategoryList
+            categories={this.state.categories}
+            setContentPane={this.setContent}
+          />
+          <AddCategoryForm addCategory={this.addCategory} />
+        </div>
+        <Category categoryName={this.state.display} />
+      </Fragment>
+    );
+  }
+}
+
+class CategoryList extends React.Component {
```

Comment:
This could also just be a function? Or a separate JS file? (Again, also if your restructuring comes in the next pr)